src/pe/certificate_table.rs

```diff
@@ -0,0 +1,146 @@
+use crate::error;
+use scroll::Pread;
+
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+#[repr(u16)]
+#[derive(Debug, PartialEq, Copy, Clone)]
+enum CertificateAttributeRevision {
+    Revision1_0 = 0x0100,
+    Revision2_0 = 0x0200,
+}
+
+impl TryFrom<u16> for CertificateAttributeRevision {
+    type Error = error::Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        Ok(match value {
+            x if x == CertificateAttributeRevision::Revision1_0 as u16 => {
+                CertificateAttributeRevision::Revision1_0
+            }
+            x if x == CertificateAttributeRevision::Revision2_0 as u16 => {
+                CertificateAttributeRevision::Revision2_0
+            }
+            _ => {
+                return Err(error::Error::Malformed(
+                    "Invalid certificate attribute revision".to_string(),
+                ))
+            }
+        })
+    }
+}
+
+#[repr(u16)]
+#[derive(Debug)]
+enum CertificateAttributeType {
+    X509 = 0x0001,
+    PkcsSignedData = 0x0002,
+    Reserved1 = 0x0003,
+    TsStackSigned = 0x0004,
```

Comment:
```suggestion
    // WIN_CERT_TYPE_X509
    X509 = 0x0001,
    // WIN_CERT_TYPE_PKCS_SIGNED_DATA
    PkcsSignedData = 0x0002,
    // WIN_CERT_TYPE_RESERVED_1
    Reserved1 = 0x0003,
    // WIN_CERT_TYPE_TS_STACK_SIGNED
    TsStackSigned = 0x0004,
```

I find putting the names used in the spec useful to try to match logic implemented here against the specification.